src/main/blackbox/blackbox.c

```diff
@@ -101,10 +101,18 @@ void checkFlashStop(void);
 #define DEFAULT_BLACKBOX_DEVICE     BLACKBOX_DEVICE_NONE
 #endif
 
-PG_REGISTER_WITH_RESET_TEMPLATE(blackboxConfig_t, blackboxConfig, PG_BLACKBOX_CONFIG, 4);
+#define FIELDS_ENABLED_DEFAULT (0 \
+    | (1 << FLIGHT_LOG_FIELD_SELECT_SETPOINT) \
+    | (1 << FLIGHT_LOG_FIELD_SELECT_GYRO) \
+    | (1 << FLIGHT_LOG_FIELD_SELECT_ACC) \
+    | (1 << FLIGHT_LOG_FIELD_SELECT_MOTOR) \
+    | (1 << FLIGHT_LOG_FIELD_SELECT_GYROUNFILT) \
+)
```

Comment:
```suggestion
#define FIELDS_ENABLED_DEFAULT (0 \
    | (1 << FLIGHT_LOG_FIELD_SELECT_SETPOINT) \
    | (1 << FLIGHT_LOG_FIELD_SELECT_GYRO) \
    | (1 << FLIGHT_LOG_FIELD_SELECT_ACC) \
    | (1 << FLIGHT_LOG_FIELD_SELECT_MOTOR) \
    | (1 << FLIGHT_LOG_FIELD_SELECT_GYROUNFILT) \
#ifdef USE_BARO
    | (1 << FLIGHT_LOG_FIELD_SELECT_BARO) \
#endif
#ifdef USE_GPS
    | (1 << FLIGHT_LOG_FIELD_SELECT_GPS) \
#endif
#ifdef USE_MAG
    | (1 << FLIGHT_LOG_FIELD_SELECT_MAG) \
#endif
#ifdef USE_SERVOS
    | (1 << FLIGHT_LOG_FIELD_SELECT_SERVOS) \
#endif
)
```

Reply:
The idea is to make the pilot actively think about what he wants to log (in addition to the basics) and therefore implicating that logging is not "for free" (needs CPU and memory resources). Automatically enabling fields based on defines defeats the purpose a bit because that's the current behavior in master.

Reply:
This suggestion is still valid for removing fields when not in build. Need to check configurator disable fields though.

Reply:
- Still have not tested yet but have a feeling this is [already] handled [partial] in

https://github.com/betaflight/betaflight/blob/59e308ae56bb33b10f7f1be289602e3ec5cc0f56/src/main/blackbox/blackbox.c#L194

- Using a preset is a good idea to opt out certain fields.